From bdc6106dbab5c4f3bd2693b54f81427a40da2d8a Mon Sep 17 00:00:00 2001
From: Jeremy Koritzinsky <jekoritz@microsoft.com>
Date: Fri, 7 Feb 2025 16:05:01 -0800
Subject: [PATCH] Set a RID for InteractiveHost32

The .NET SDK's RID (and PlatformTarget) inference for .NET Framework only executes on Windows. As a result, building InteractiveHost32 on non-Windows produces a differently-named NuGet package. This is causing us issues in the VMR.

Set the RID explicitly like the InteractiveHost64 process does. This ensures that the resulting package name is InteractiveHost32.x86.Symbols when built on Windows or non-Windows.

Backport: https://github.com/dotnet/roslyn/pull/77116
---
 src/Interactive/HostProcess/x86/InteractiveHost32.csproj | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Interactive/HostProcess/x86/InteractiveHost32.csproj b/src/Interactive/HostProcess/x86/InteractiveHost32.csproj
index 70e32290b8987..3f5e9f3862b8e 100644
--- a/src/Interactive/HostProcess/x86/InteractiveHost32.csproj
+++ b/src/Interactive/HostProcess/x86/InteractiveHost32.csproj
@@ -9,6 +9,7 @@
     <UseWindowsForms>true</UseWindowsForms>
     <ExcludeFromSourceOnlyBuild>true</ExcludeFromSourceOnlyBuild>
     <IsSymbolPublishingPackage>true</IsSymbolPublishingPackage>
+    <RuntimeIdentifier>win-x86</RuntimeIdentifier>
   </PropertyGroup>
   <ItemGroup>
     <ProjectReference Include="..\..\..\Compilers\Core\Portable\Microsoft.CodeAnalysis.csproj" />
